src/test/java/raisetech/StudentManagement/service/StudentServiceTest.java

```diff
@@ -0,0 +1,111 @@
+import raisetech.StudentManagement.controller.converter.StudentConverter;
+import raisetech.StudentManagement.data.Student;
+import raisetech.StudentManagement.data.StudentCourse;
+import raisetech.StudentManagement.domain.StudentDetail;
+import raisetech.StudentManagement.repositry.StudentRepository;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class StudentServiceTest {
+
+    @Mock
+    private StudentRepository repository;
+
+    @Mock
+    private StudentConverter converter;
+
+    private StudentService sut;
+
+    @BeforeEach
+    void before() {
+        sut = new StudentService(repository, converter);
+    }
+
+
+    @Test
+    void 受講生詳細の一覧検索_リポジトリとコンバーターの処理が適切に呼び出せていること() {
+        ;
+        List<Student> studentList = new ArrayList<>();
+        List<StudentCourse> studentCourseList = new ArrayList<>();
+        when(repository.search()).thenReturn(studentList);
+        when(repository.searchStudentCoursesList()).thenReturn(studentCourseList);
+
+        sut.searchStudentList();
+
+
+        Mockito.verify(repository, Mockito.times(1)).search();
+        Mockito.verify(repository, Mockito.times(1)).searchStudentCoursesList();
+        Mockito.verify(converter, Mockito.times(1)).convertStudentDetails(studentList, studentCourseList);
+
+
+        @Test
+        void 受講生詳細の登録_リポジトリの処理が適切に呼び出せていること () {
+            String id = "999";
+            Student student = new Student();
+            student.setId(id);
+            when(repository.searchStudent(id)).thenReturn(student);
+            when(repository.searchStudentCourse(id)).thenReturn(new ArrayList<>());
+
+            StudentDetail expected = new StudentDetail(student, new ArrayList<>());
+            StudentDetail actual = sut.searchStudent(id);
+
+            verify(repository, times(1)).searchStudent(id);
+            verify(repository, times(1)).searchStudentCourse(id);
+            Assertions.assertEquals(expected.getStudent().getId(), actual.getStudent().getId());
+        }
+
+        @Test
+        void 受講生の登録処理_リポジトリのメソッドが適切に呼び出されること () {
+            Student student = new Student();
+            StudentCourse studentCourse = new StudentCourse();
+            List<StudentCourse> studentCourseList1 = List.of(studentCourse);
+
+            sut.registerStudent(student);
+            sut.registerStudentCourse(studentCourse);
+
+            verify(repository, times(1)).registerStudent(student);
+            verify(repository, times(1)).registerStudentCourse(studentCourse);
+        }
+
+        @Test
+        void 受講生詳細の登録_初期化処理が行われること () {
+            String id = "999";
+            Student student = new Student();
+            student.setId(id);
+            StudentCourse studentCourse = new StudentCourse();
+
+            sut.initStudentcoursse(studentCourse, student.getId());
```

Comment:
メソッド名の方を修正して欲しいのですが、typoとキャメルケースになっていないですね
```suggestion
            sut.initStudentCourse(studentCourse, student.getId());
```